### PR TITLE
plonk book library section

### LINKED ISSUE #23: Plonk Book / Implementation Documentation
Add an extensive documentation in the style of [Halo2 Book](https://zcash.github.io/halo2/index.html).
It should contain:

- Theoretical background and explanation of Plonk
- Details of the implementation including: description of the structures and traits, complexitites of the main operations, differences from the original protocol, etc.
- Examples of use

The purpose of the book should be to allow new developers to learn how to use this library in the easiest way possible.

### PR DESCRIPTION
**Pull request by markulf**
general writing improvements

### CODE REVIEW

File: plonk-book/src/chapter_3.md [hunk 1]
```diff
@@ -60,11 +60,9 @@ In order to use these gates only when needed we introduce selector polynomials (
 The gate constraint equation is represented as follow:
 
 $$
-q_{arith} \cdot arith\_eq +
-q_{logic} \cdot  logic\_eq +
-q_{range} \cdot  range\_eq +
-q_{ECfix} \cdot   ECfix\_eq +
-q_{ECvar} \cdot  ECvar\_eq
+q_{arith} \cdot \textit{arith\_eq} + q_{logic} \cdot  \textit{logic\_eq} + q_{range} \cdot  \textit{range\_eq} +
```
Comment (markulf) [inline plonk-book/src/chapter_3.md:62]: Does such a selector polynomial really exist, or is it split up into $q_l, q_r, q_o, q_m, q_c$

I see now that we have both types, myabe $q_{arith}$ is hierarchically at a higher level. Can we explain this better?

Comment (davidnevadoc) [reply plonk-book/src/chapter_3.md:62]: We can consider that `q_arith` is hierarchically at a higher level. 
`q_l, q_r, q_c, q_m, q_o, q_4` could be considered wire selectors in arithmetic gates, and auxiliary values in other custom gates.
On the other hand `q_arith, q_logic, ...` are gate selectors which determine which gate is activated at any point in the circuit.

The explanation should be expanded.

Comment (mathcrypto) [reply plonk-book/src/chapter_3.md:62]: Ok will expand the explanation

**Review by mathcrypto — APPROVED** [review]

File: plonk-book/src/chapter_3.md (hunk 1 above)
Comment (markulf) [reply plonk-book/src/chapter_3.md:62]: Great, that makes sense. Shall we do this in a separate pull request. I created an issue to keep track of this idea.

**Comment by markulf** [comment]
Ready to squash and merge?

**Comment by mathcrypto** [comment]
> Ready to squash and merge?

LGTM, @davidnevadoc is it ok with you?

**Review by davidnevadoc — APPROVED** [review]
Good improvements, everything looks good! :+1:

**Pull request merged**